homeassistant/components/squeezebox/__init__.py

```diff
@@ -73,18 +89,41 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
     )
     _LOGGER.debug("LMS %s = '%s' with uuid = %s ", lms.name, host, lms.uuid)
 
-    entry.runtime_data = lms
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, lms.uuid)},
+        name=lms.name,
+        manufacturer=MANUFACTURER,
+        model=SERVER_MODEL,
+        sw_version=STATUS_QUERY_VERSION in status
+        and status[STATUS_QUERY_VERSION]
+        or None,
+        serial_number=lms.uuid,
+        connections=STATUS_QUERY_MAC in status
+        and {(CONNECTION_NETWORK_MAC, format_mac(status[STATUS_QUERY_MAC]))}
+        or None,
```

Comment:
Can we get the sw version and connections in a variable beforehand? This looks very messy

Reply:
yes agreed messay wasn't till ruff got hold of it.